kernel/main.c

```diff
@@ -130,8 +130,8 @@ struct thread_info *start_kernel(void)
     show_page_bitmap();  // init_pages();
     kmem_cache_init();
 
-    /* select a scheduling policy */
-    sched_select(SCHED_CLASS_BITMAP);
+    /* initialize scheduler with specified scheduling policy */
+    sched_init(SCHED_CLASS_BITMAP);
```

Comment:
It is not trivial to distinguish from 2 cases of function `sched_init`:
* initialize scheduler internals in general; or
* initialize given scheduler for each.

The misleading semantics prevent us from fair usage.

Reply:
I think this isn't  good. `sched_init` should do the general part of initializing scheduler, not specified scheduling policy.

For example, it would be somthing like:

```
/* initialize each scheduling policy */
sched_init();

void sched_init(void) {
    sched_rr->init();
    sched_bitmap->init();
}
```

Reply:
Continuing feedback from @mlouielu, I would expect the availability of scheduler policy implemented listed via certain "hooks" as if shell commands are registered while sched_init() is concerned.